README.md

```diff
@@ -31,3 +31,39 @@ func main() {
 	function.Run(hello)
 }
 ```
+
+# Local Development
+
+You can debug your function locally by calling `function.Debug()` instead of `function.Run()` from within a test.
+
+```go
+// main_test.go
+package main
+
+import (
+	"testing"
+	"github.com/spiceai/spice-functions-go/function"
+)
+
+func TestHello(t *testing.T) {
+	// Debug returns the DuckDB database that was used in the function, as well as any errors.
+	// Close the database when you're done with it.
+	outputDb, err := function.Debug(hello, function.WithApiKey("<your-api-key>"))
+	if err != nil {
+		t.Error(err)
+	}
+	defer outputDb.Close()
+
+	// Use the outputDb to query the results of your function.
+}
+```
+
+You can pass a number of options to `function.Debug()` to configure the function's execution.
+
+- `function.WithApiKey("<spice-api-key>")`: **Required**. Sets the Spice API key to use when running the function. Defaults to the `SPICE_API_KEY` environment variable.
```

Comment:
Not sure that the With notation makes sense for essentially required config here. For the optional inputs, maybe makes sense, but it kinda feels like we're forcing this into the With pattern?